wasm-wasi-core/src/desktop/process.ts

```diff
@@ -104,14 +104,19 @@ export class NodeWasiProcess extends WasiProcess {
 		return Promise.resolve();
 	}
 
+	protected async cleanupResources(): Promise<void> {
+		await this.mainWorker?.terminate();
```

Comment:
This is the only part I'm unsure of. I did not find an easy way to both allow the base class to terminate the `mainWorker` to clean the resource, and also allow the `terminate` method to get the result from termination. So for now, I've duplicated the calls assuming calling terminate twice shouldn't be an issue.